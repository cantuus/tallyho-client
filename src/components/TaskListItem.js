import React, { Component } from 'react'
//import TallyhoApiService from '../services/tallyho-api-service'

export default class TaskListItem extends Component {
    render() {
        return ( 
            <section>
                <h3 className="task-title">{this.props.title}</h3>
                <img src={this.props.image} alt={`task of ${this.props.image}`} />
            </section>
        )
    }
}