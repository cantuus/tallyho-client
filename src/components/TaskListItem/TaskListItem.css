.task-item {
    background-color: #ff8a5c;
    padding-top: 1rem;
    padding-bottom: 1rem;
    margin-bottom: 1rem;
}

.task-image {
    max-height:150px;
}

.task-image-checked {
    max-height: 150px;
}

.task-delete {
    display: flex;
    justify-content: flex-end;
    margin-top: 1rem;
    margin-right: 2rem;
}

.fa-trash-alt:hover {
    color:#fcf9ea;
}

/* Medium devices (tablets, 768px and up) */
@media screen and (min-width: 800px) {
    .task-form-page {
        margin-left: -4rem;
        margin-right: -3rem;
    }

}

/*  Large devices (desktops, 992px and up) */
@media screen and (min-width: 992px) {
    .task-image {
        max-height: 14rem;
    }
}


/* Extra large devices (large desktops, 1200px and up) */
@media (min-width: 1200px) { 
    .task-image {
        max-height: 14rem;
    }
}