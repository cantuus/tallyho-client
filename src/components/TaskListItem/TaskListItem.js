import React, { Component } from 'react'
import TallyhoApiService from '../../services/tallyho-api-service'

export default class TaskListItem extends Component {
    constructor(props) {
        super(props)
        this.handleClickDelete = this.handleClickDelete.bind(this)
    }

    state = {
        checked: this.props.task.checked,
        image: this.props.task.image
    }

    handleClickDelete = () => {
        const taskId = this.props.task.id;
        const thisProps = this.props;
        console.log(thisProps);

        TallyhoApiService.getTaskToDelete(taskId)
            .then(() => {
                this.props.handleDeleteTask(taskId)
            })
            .catch(error => {
                console.error({ error })
            })


    }

    handleClickToggle = () => {
        let newTask = this.props.task;
        newTask.checked = !newTask.checked;

        TallyhoApiService.updateTask(newTask, newTask.id)
            .then(() => {
                this.setState({
                    checked: !this.state.checked
                })
            })
            .catch(error => {
                console.error({ error })
            })
    }

    changeImage = () => {
        let imageUrl;

        if (this.state.checked) {
             imageUrl = 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTQ-kMbFqjiAXzJ3W9-ODBgavLSogiAblyrWMF5ia7HfQBAt3Qy&s'
        }
        else {
            imageUrl = this.props.task.image;
        }

        return imageUrl
    }


    render() {
        return (
            <section>
                <h3 className="task-title">{this.props.task.title}</h3>
                <img onClick={this.handleClickToggle} src={this.changeImage()} alt={`task of ${this.props.task.image}`} />
                <button onClick={() => this.handleClickDelete()}>Delete</button>
            </section>
        )
    }
}

//src=use a terniary to render image based on the checked value
