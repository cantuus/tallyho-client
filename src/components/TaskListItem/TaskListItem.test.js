import React from 'react'
import ReactDOM from 'react-dom'
import TaskListItem from './TaskListItem'
import { shallow, mount } from 'enzyme'

const handleClickDelete = jest.fn();
const changeImage = jest.fn();
const changeImageClass = jest.fn();

describe.only('TaskListItem', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        const testTask = [
            {
                title: 'test title',
                image: 'https//testimage.com',
                checked: false,
                user_id: 1
            }
        ]
        ReactDOM.render(<TaskListItem task={testTask} />, div);

        ReactDOM.unmountComponentAtNode(div)
    })

    it('should render title inside of h3', () => {
        let props = {
            title: 'test title',
            image: 'https//testimage.com',
            checked: false,
            user_id: 1
        };

        const wrapper = shallow(<TaskListItem task={props} />);
        const title = wrapper.find('h3');
        expect(title).toHaveLength(1);
        expect(title.text()).toEqual('test title');

    })

    it('test delete button', () => {
        let props = {
            title: 'test title',
            image: 'https//testimage.com',
            checked: false,
            user_id: 1
        };

        //reset info from previous calls - particularly toggles!
        // handleClickDelete.mockClear()

        const wrapper = mount(
            <TaskListItem
                task={props}
                onClick={handleClickDelete}
            />);

        wrapper.find("div").simulate("click");

        expect(handleClickDelete).toHaveBeenCalled();

    });

    it('show checkmark image when checked is true', () => {
        let props = {
            title: 'test title',
            image: 'https//testimage.com',
            checked: true,
            user_id: 1
        };

        const wrapper = shallow(
            <TaskListItem
                task={props}
            />)

        expect(wrapper).toMatchSnapshot();
    });

    it('show task image when checked is false ', () => {
        let props = {
            title: 'test title',
            image: 'https//testimage.com',
            checked: false,
            user_id: 1
        };

        const wrapper = shallow(
            <TaskListItem
                task={props}
            />)

        expect(wrapper).toMatchSnapshot();

    })
})