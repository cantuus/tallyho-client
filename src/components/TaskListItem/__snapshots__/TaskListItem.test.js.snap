// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListItem show checkmark image when checked is true 1`] = `ShallowWrapper {}`;

exports[`TaskListItem show task image when checked is false  1`] = `ShallowWrapper {}`;
